### PR TITLE
Update Cobalt Android NDK to 25.2

### PR DESCRIPTION
**Pull request by zhongqiliang**
b/287076646

This is based on Kaido's change https://github.com/youtube/cobalt/pull/596

### CODE REVIEW

File: starboard/android/shared/platform_configuration/BUILD.gn [hunk 1]
```diff
@@ -112,6 +112,17 @@ config("platform_configuration") {
 
     # Don"t warn for implicit sign conversions. (in v8 and protobuf)
     "-Wno-sign-conversion",
+
+    # All these below should be fixed
```
[the remark this answers is not in the document]
Comment (zhongqiliang) [reply starboard/android/shared/platform_configuration/BUILD.gn:116]: Do we want to create bugs to clean up this warnings in the code? @kaidokert

**Comment by zhongqiliang** [comment]
nplb tests failed on both the main branch and this branch locally.

File: starboard/android/shared/platform_configuration/BUILD.gn (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaidokert) [reply starboard/android/shared/platform_configuration/BUILD.gn:116]: I would try and clean up the code right away if you could, for the ones that only trigger in one or two lines of code.

For the ones that generate hundreds of warnings, it's better to create a follow-up bug. 

( for the record when i made this as a sample, i just went very fast and didn't have a local build config set up as well, that's why i didn't look at addressing them individually ).

[the remark this answers is not in the document]
Comment (zhongqiliang) [reply starboard/android/shared/platform_configuration/BUILD.gn]: I have done the investigation, and added the TODO bugs, they are in this hotlist. https://b.corp.google.com/hotlists/5084470

File: starboard/android/shared/platform_configuration/BUILD.gn [hunk 2]
```diff
@@ -112,6 +112,15 @@ config("platform_configuration") {
 
     # Don"t warn for implicit sign conversions. (in v8 and protobuf)
     "-Wno-sign-conversion",
+
+    # (TODO)b/287129945
```
[the remark this answers is not in the document]
Comment (zhongqiliang) [reply starboard/android/shared/platform_configuration/BUILD.gn]: Done

File: starboard/android/shared/test_filters.py [hunk 3]
```diff
@@ -72,6 +72,12 @@
         # when invalid initialization data is passed to
         # SbDrmGenerateSessionUpdateRequest().
         'SbDrmSessionTest.InvalidSessionUpdateRequestParams',
+
+        # This test crashed after NDK 25 upgrade.
```
Comment (kaidokert) [inline starboard/android/shared/test_filters.py:76]: Please try to keep tracking bugs for disabled tests in the comment lines here. With our current setup it's easier to manage the follow-ups and cleanups if they are logged as bugs.

I know this file doesn't have them, but we have tried for other platforms ( e.g. go/cobalt-cl/252900 )

File: third_party/zlib/contrib/minizip/zip.c [hunk 4]
```diff
@@ -537,8 +537,8 @@ local ZPOS64_T zip64local_SearchCentralDir(const zlib_filefunc64_32_def* pzlib_f
         break;
       }
 
-      if (uPosFound!=0)
-        break;
+    if (uPosFound!=0)
```
Comment (kaidokert) [inline third_party/zlib/contrib/minizip/zip.c:540]: Thanks for taking care of this ! Small thing, only one instance

**Review by kaidokert — APPROVED** [review]
lgtm to go in.

Please remember the follow-ups when merging: Internal CI containers need to be re-deployed, otherwise the build containers will have old NDK in them.

And please add bugs to disabled tests

File: third_party/zlib/contrib/minizip/zip.c (hunk 4 above)
Comment (zhongqiliang) [reply third_party/zlib/contrib/minizip/zip.c:540]: ack

**Pull request merged**